code42_connector.py

```diff
@@ -0,0 +1,156 @@
+# Python 3 Compatibility imports
+from __future__ import print_function, unicode_literals
+
+# Phantom App imports
+import phantom.app as phantom
+from phantom.base_connector import BaseConnector
+from phantom.action_result import ActionResult
+
+# Usage of the consts file is recommended
+# from code42_consts import *
+import requests
+import json
+
+import py42.sdk
+import py42.util as util
+
+
+class RetVal(tuple):
+
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class Code42Connector(BaseConnector):
+
+    def __init__(self):
+        super(Code42Connector, self).__init__()
+
+        self._state = None
+        self._cloud_instance = None
+        self._username = None
+        self._password = None
+        self._client = None
+
+    def _create_client(self, action_result):
+        try:
+            self._client = py42.sdk.from_local_account(self._cloud_instance, self._username, self._password)
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, f"Could not create py42 client: {e}")
+
+        return phantom.APP_SUCCESS
+
+    def _handle_test_connectivity(self, param):
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        self.save_progress("Connecting to endpoint")
+
+        if not self._create_client(action_result):
```

Comment:
would it make sense for us to call `_create_client` at the end of `initialize()` so that we won't have to include this little snippet in every single action handler?

Reply:
I'm assuming `initialize()` gets called automatically before every action through some kind of black magic because I don't see it being called anywhere in this code.

Reply:
I think I figured this out, `main()` calls `_handle_action()`, which must then call `initialize()` followed by `handle_action()`.

Reply:
That's a good point. I modeled the client creation piece after the AWS apps I found [here](https://github.com/phantomcyber/phantom-apps/blob/next/Apps/phawscloudtrail/awscloudtrail_connector.py#L171), but I tried it the way you suggested and it appears to work just fine. I'm inclined towards calling it in `initialize()` like you suggest to reduce code duplication.